include/cantera/kinetics/ReactionRate.h

```diff
@@ -206,6 +228,102 @@ class ReactionRate : public ReactionRateBase
     }
 };
 
+//! Class special template for reaction rate definitions with ETempData
+/**
+ * This class template ensures that derived objects are aware of specialized
+ * data types, which are passed by MultiRateBase evaluators.
+ */
+template <>
+class ReactionRate<ETempData> : public ReactionRateBase
```

Comment:
This class shouldn't be necessary if you remove all specialized methods for `Te`.